keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -506,6 +509,30 @@ With recent conversations about kube-apiserver and enabled featureset version,
 it is possible to incorporate other version relating to the cluster, such as
 minimum kubelet version, maximum kubelet version, and enabled featureset version.
 
+#### Credentials
+
+An array of credentials to access the cluster. Each credential should be assigned 
+to a different consumer. Each credential has: 
+- a **consumer** field to indicate which consumer it is assigned to.
+- a **context** field to indicate the context name in the access information that should be used to access the cluster.
+- an **accessRef** field to point to the access information location. 
+
+The access information itself is stored in the same Kubernetes cluster on which
+the ClusterProfile API resides. However, the access information for different 
+consumers is stored in different namespaces to ensure that it is only accessible
+to the consumer for which it is intended. We recommend that the access information
+be stored in a secret object to add further protection to the access information.
+
+##### Access information format
+The access information must contain the following fields
+- **Config**: This field contains cluster access information compatible with the 
+  [kubeconfig format](https://github.com/kubernetes/kubernetes/blob/dbc2b0a5c7acc349ea71a14e49913661eaf708d2/staging/src/k8s.io/client-go/tools/clientcmd/api/types.go#L31).
+
+Please note that a single [Kubeconfig file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/)
+supports access to multiple clusters. Consequently, multiple `ClusterProfile`
+objects can share the same access secret. However, it is crucial to
+ensure that each secret contains access information for only a single consumer API.
```

Comment:
This is a good idea, but how do we enforce it? Is it just based on the object name specified in the "consumer" field?

Reply:
+1 on always make it a Secret since it's meant for storing sensitive data like this.

You are right Babak. It's hard to fully prevent one consumer from using credentials meant for another just by using the credentialConsumer field. It helps label them, but doesn't provide any actual enforcement. If that's a concern, providers are expected to place ClusterProfiles in separate namespaces so they have different access creds in each. That way namespace boundaries help enforce who can access which credentials.

Reply:
> Consequently, multiple `ClusterProfile` objects can share the same access secret.

Wouldn't it be simpler to just disallow this sharing?

Reply:
> Wouldn't it be simpler to just disallow this sharing?

I think that's actually fine. In a lot of cases, I can imagine that a consumer could leverage the same credentials to reach all member clusters, in that case, leveraging the same secret makes management a lot easier and security is actually improved (over having multiple files which may make it more error prone given their multitude)